spot-contracts/contracts/PerpetualTranche.sol

```diff
@@ -911,63 +823,47 @@ contract PerpetualTranche is
             return r;
         }
 
         //-----------------------------------------------------------------------------
         // Basic rollover with fees:
-        // (1 +/- f) . (stdTrancheInAmt . trancheInPrice) = (stdTrancheOutAmt . trancheOutPrice)
+        // (1 +/- f) . (trancheInAmt . trancheInPrice) = (tokenOutAmt . tokenOutPrice)
         //-----------------------------------------------------------------------------
 
-        // Given the amount of tranches In, we compute the amount of tranches out
+        // Given the amount of tranches In, we compute the amount of tokens out
         r.trancheInAmt = trancheInAmtAvailable;
-        uint256 stdTrancheInAmt = _toStdTrancheAmt(trancheInAmtAvailable, trancheInDiscount);
-        uint256 stdTrancheOutAmt = stdTrancheInAmt.mulDiv(trancheInPrice, trancheOutPrice);
+        r.tokenOutAmt = r.trancheInAmt.mulDiv(trancheInPrice, tokenOutPrice);
 
-        // A positive fee percentage implies that perp charges rotators by
-        // accepting tranchesIn at a discount, ie) fewer tranches out.
+        // A postive fee percentage implies that perp charges rotators by
+        // accepting tranchesIn at a discount, ie) fewer tokens out.
         if (feePerc > 0) {
-            stdTrancheOutAmt = stdTrancheOutAmt.mulDiv(HUNDRED_PERC - feePerc.abs(), HUNDRED_PERC);
+            r.tokenOutAmt = r.tokenOutAmt.mulDiv(HUNDRED_PERC - feePerc.abs(), HUNDRED_PERC);
         }
         // A negative fee percentage (or a reward) implies that perp pays the rotators by
-        // accepting tranchesIn at a premium, ie) more tranches out.
+        // accepting tranchesIn at a premium, ie) more tokens out.
         else if (feePerc < 0) {
-            stdTrancheOutAmt = stdTrancheOutAmt.mulDiv(HUNDRED_PERC + feePerc.abs(), HUNDRED_PERC);
+            r.tokenOutAmt = r.tokenOutAmt.mulDiv(HUNDRED_PERC + feePerc.abs(), HUNDRED_PERC);
         }
-        r.trancheOutAmt = _fromStdTrancheAmt(stdTrancheOutAmt, trancheOutDiscount, MathUpgradeable.Rounding.Down);
         //-----------------------------------------------------------------------------
 
-        // However, if the tokenOut is the mature tranche (held as naked collateral),
-        // we infer the tokenOut amount from the tranche denomination.
-        // (tokenOutAmt = collateralBalance * trancheOutAmt / matureTrancheBalance)
-        bool isMatureTrancheOut = _isMatureTranche(tokenOut);
-        r.tokenOutAmt = isMatureTrancheOut
-            ? tokenOutBalance.mulDiv(r.trancheOutAmt, _matureTrancheBalance)
-            : r.trancheOutAmt;
-
         // When the tokenOut balance is NOT covered:
         // we fix tokenOutAmt = tokenOutAmtRequested and re-calculate other values
-        if (r.tokenOutAmt > tokenOutAmtRequested) {
-            // Given the amount of tranches out, we compute the amount of tranches in
+        if (r.tokenOutAmt >= tokenOutAmtRequested) {
```

Comment:
Is the `=` in`>=` correct? Sounds like we need to continue only if there'll be tokens left over

Reply:
Right we don't need to re-compute when `r.tokenOutAmt == tokenOutAmtRequested`